nuntium/models.py

```diff
@@ -303,7 +303,17 @@ def moderate(self):
             raise ValidationError(_('The message needs '
                 + 'to be confirmated first'))
         self.set_to_ready()
-        self.moderation.success()
+        # if we turn on moderation after some messages have been created then
+        # they will not have associated Moderation objects so we need to catch
+        # that, create the moderation object and then try again.
+        try:
+            self.moderation.success()
+        except Moderation.DoesNotExist:
+            if self.writeitinstance.config.moderation_needed_in_all_messages:
+                moderation, created = Moderation.objects.get_or_create(message=self)
+                if created:
+                    return self.moderation.success()
```

Comment:
Since the `return` here is just for control flow rather than explicitly returning a `None`, I'd split it into:
```
self.moderation.success()
return
```